src/compiler/tsbuildPublic.ts

```diff
@@ -1459,80 +1506,182 @@ namespace ts {
+            // Collect the expected outputs of this project
+            const outputs = getAllProjectOutputs(project, !host.useCaseSensitiveFileNames());
+            const outputTimeStampMap = getOutputTimeStampMap(state, resolvedPath);
+            for (const output of outputs) {
+                const path = toPath(state, output);
+                // Output is missing; can stop checking
+                let outputTime = outputTimeStampMap?.get(path);
+                if (!outputTime) {
+                    outputTime = ts.getModifiedTime(state.host, output);
+                    outputTimeStampMap?.set(path, outputTime);
+                }
+
+                if (outputTime === missingFileModifiedTime) {
+                    return {
+                        type: UpToDateStatusType.OutputMissing,
+                        missingOutputFileName: output
+                    };
+                }
+
+                if (outputTime < oldestOutputFileTime) {
+                    oldestOutputFileTime = outputTime;
+                    oldestOutputFileName = output;
+                }
+
+                // If an output is older than the newest input, we can stop checking
+                if (outputTime < newestInputFileTime) {
+                    return {
+                        type: UpToDateStatusType.OutOfDateWithSelf,
+                        outOfDateOutputFileName: oldestOutputFileName,
+                        newerInputFileName: newestInputFileName
+                    };
+                }
+            }
         }
 
-        if (!force && !state.buildInfoChecked.has(resolvedPath)) {
-            state.buildInfoChecked.set(resolvedPath, true);
-            const buildInfoPath = getTsBuildInfoEmitOutputFilePath(project.options);
-            if (buildInfoPath) {
-                const value = state.readFileWithCache(buildInfoPath);
-                const buildInfo = value && getBuildInfo(value);
-                if (buildInfo && (buildInfo.bundle || buildInfo.program) && buildInfo.version !== version) {
+        const seenRefs = buildInfoPath ? new Set<ResolvedConfigFilePath>() : undefined;
+        const buildInfoCacheEntry = state.buildInfoCache.get(resolvedPath)!;
+        seenRefs?.add(resolvedPath);
+
+        let pseudoUpToDate = false;
+        let usesPrepend = false;
+        let upstreamChangedProject: string | undefined;
+        if (referenceStatuses) {
+            for (const { ref, refStatus, resolvedConfig, resolvedRefPath } of referenceStatuses) {
+                usesPrepend = usesPrepend || !!(ref.prepend);
+                // If the upstream project's newest file is older than our oldest output, we
+                // can't be out of date because of it
+                if (refStatus.newestInputFileTime && refStatus.newestInputFileTime <= oldestOutputFileTime) {
+                    continue;
+                }
+
+                // Check if tsbuildinfo path is shared, then we need to rebuild

```

Comment:
This is specific to test we have to check if referencing project has same tsbuildInfo we need to build and report error. Earlier it use to fail because we would do output timestamp check, but since we dont do that anymore, we need to check  it specifically here